hips/hip-00xx-sequencing.md

```diff
@@ -0,0 +1,192 @@
+- `helm.sh/depends-on/layer`: Declare a layer that must exist and in a ready state before this resource can be deployed.
+- `helm.sh/depends-on/chart`: Declare a chart dependency that must exist and in a ready state before this resource can be deployed. For the chart to be declared ready, all of its resources, with their sequencing order taken into consideration, would need to be deployed and declared ready.
+
+A layer would not be considered "ready" and the next layer deployed until all resources in that layer are considered "ready". Readiness is described in a later section.
+
+#### Example:
+```yaml
+# resource 1
+metadata:
+  name: foo
+  annotations:
+    helm.sh/layer: layer1
+---
+# resource 2
+metadata:
+  name: bar
+  annotations:
+    helm.sh/layer: layer2
+    helm.sh/depends-on/layer: layer1
+---
+# resource 3
+metadata:
+  name: fizz
+  annotations:
+    helm.sh/layer: layer3
+    helm.sh/depends-on/layer: layer1
+    helm.sh/depends-on/chart: someChartDependency
+```
+In this example, Helm would be responsible for resolving the annotations on these three resources and deploy all resources in the following order:
+
+```
+layer1:    [foo]
+             ||
+             ||   [someChartDependency]
+             ||     ||
+             \/     ||
+layer2:    [bar]    ||
+              \\   //
+               \/ \/
+layer3:       [fizz]
+```
+
+This approach is prone to circular dependencies. During the templating phase, Helm will have logic to detect, and report any circular dependencies found in the chart templates.
+
+### Proposal 2: Weighted Resouces (without hooks)
+
+Expand on the the weight capabilities of hooks, but allow them to be defined outside of the context of hooks for the chart. Add a new annotation `helm.sh/weight` that helm then uses to order when resources are deployed based on the same weighting rules that hooks use today.
+
+#### Example:
+```yaml
+# resource 1
+metadata:
+  name: foo
+  annotations:
+    helm.sh/weight: "-2"
+---
+# resource 2
+metadata:
+  name: bar
+  annotations:
+    helm.sh/weight: "-1"
+---
+# resource 3
+metadata:
+  name: fizz
+  annotations:
+    helm.sh/weight: 0
+```
+In this example, Helm would be responsible for resolving the annotations on these three resources and deploy all resources in the following order:
+
+```
+-2:    [foo]
+        ||
+        \/
+-1:    [bar]
+        ||
+        \/
+0:     [fizz]
+```
+
+### Readiness
+
+In order to enforce sequencing, new `helm.sh/resource-ready` annotation would be used to determine when a resource is ready, allowing helm to proceed deploying the next group of resources, or failing a deployment. Helm would query, using jsonpath syntax, status fields of the annotated resource. Some native kubernetes resources that have stable APIs e.g `v1` resources such as `Pod` and `Deployment`, would have default queries which can be overriden. In Helm cannot determine how to check a resource's readiness when it should, it will do nothing and log this.
```

Comment:
guessing you meant "if" Helm can not determine readiness?
```suggestion
In order to enforce sequencing, new `helm.sh/resource-ready` annotation would be used to determine when a resource is ready, allowing helm to proceed deploying the next group of resources, or failing a deployment. Helm would query, using jsonpath syntax, status fields of the annotated resource. Some native kubernetes resources that have stable APIs e.g `v1` resources such as `Pod` and `Deployment`, would have default queries which can be overriden. If Helm cannot determine how to check a resource's readiness when it should, it will do nothing and log this.
```